### PR TITLE
[frontend-api] Added special article resolvers

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| In SSFW there are some special articles displayed in different places than common articles (e.g. cookies). This PR adds new methods for retrieving these articles.
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/frontend-api/apiary.apib [hunk 1]
```diff
@@ -2233,3 +2233,114 @@ Returns article filtered using UUID
...
+                "name": "Privacy policy",
+                "text": "Morbi posuere mauris dolor, quis accumsan dolor ullamcorper eget. Phasellus at elementum magna, et pretium neque. Praesent tristique lorem mi, eget varius quam aliquam eget. Vivamus ultrices interdum nisi, sed placerat lectus fermentum non. Phasellus ac quam vitae nisi aliquam vestibulum. Sed rhoncus tortor a arcu sagittis placerat. Nulla lectus nunc, ultrices ac faucibus sed, accumsan nec diam. Nam auctor neque quis tincidunt tempus. Nunc eget risus tristique, lobortis metus vitae, pellentesque leo. Vivamus placerat turpis ac dolor vehicula tincidunt. Sed venenatis, ante id ultrices convallis, lacus elit porttitor dolor, non porta risus ipsum ac justo. Integer id pretium quam, id placerat nulla.",
+                "seoH1": null,
+                "seoTitle": null,
+                "seoMetaDescription": null
+            }
+        }
+    }
+
+## Article detail [/graphql{?cookiesArticle}]
```
Comment (TomasLudvik) [inline packages/frontend-api/apiary.apib:2311]:
```suggestion
## Cookies article [/graphql{?cookiesArticle}]
```

File: packages/frontend-api/apiary.apib [hunk 2]
```diff
@@ -2233,3 +2233,114 @@ Returns article filtered using UUID
...
+
+### POST [POST]
+
+Returns Cookie article
```
Comment (TomasLudvik) [inline packages/frontend-api/apiary.apib:2315]:
```suggestion
Returns article detail
```

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php [hunk 3]
```diff
@@ -54,13 +63,64 @@ public function resolver(string $uuid): Article
...
+     * @return \Shopsys\FrameworkBundle\Model\Article\Article
+     */
+    public function privacyPolicyArticle(): Article
+    {
+        try {
+            $articleId = $this->setting->getForDomain(
+                Setting::PRIVACY_POLICY_ARTICLE_ID,
+                $this->domain->getId()
+            );
+            return $this->articleFacade->getById($articleId);
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php:92]: LegalConditionsFacade::findPrivacyPolicy() should be used here

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php [hunk 4]
```diff
@@ -54,13 +63,64 @@ public function resolver(string $uuid): Article
...
+     * @return \Shopsys\FrameworkBundle\Model\Article\Article
+     */
+    public function cookiesArticle(): Article
+    {
+        try {
+            $articleId = $this->setting->getForDomain(
+                Setting::COOKIES_ARTICLE_ID,
+                $this->domain->getId()
+            );
+            return $this->articleFacade->getById($articleId);
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php:108]: CookiesFacade::findCookiesArticleByDomainId() should be used here

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php [hunk 5]
```diff
@@ -54,13 +63,64 @@ public function resolver(string $uuid): Article
...
+     * @return \Shopsys\FrameworkBundle\Model\Article\Article
+     */
+    public function termsAndConditionsArticle(): Article
+    {
+        try {
+            $articleId = $this->setting->getForDomain(
+                Setting::TERMS_AND_CONDITIONS_ARTICLE_ID,
+                $this->domain->getId()
+            );
+            return $this->articleFacade->getById($articleId);
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php:76]: LegalConditionsFacade::findTermsAndConditions() should be used here

File: packages/frontend-api/src/Resources/config/graphql-types/QueryDecorator.types.yaml [hunk 6]
```diff
@@ -83,3 +83,15 @@ QueryDecorator:
                     uuid:
                         type: "Uuid!"
                 description: "Returns article filtered using UUID"
+            termsAndConditionsArticle:
+                type: 'Article'
+                resolve: "@=resolver('termsAndConditionsArticle')"
+                description: "Returns Terms and Conditions article"
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Resources/config/graphql-types/QueryDecorator.types.yaml:89]:
```suggestion
                description: "Returns terms and conditions article"
```

File: packages/frontend-api/src/Resources/config/graphql-types/QueryDecorator.types.yaml [hunk 7]
```diff
@@ -83,3 +83,15 @@ QueryDecorator:
...
+            privacyPolicyArticle:
+                type: 'Article'
+                resolve: "@=resolver('privacyPolicyArticle')"
+                description: "Returns Privacy Policy article"
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Resources/config/graphql-types/QueryDecorator.types.yaml:93]:
```suggestion
                description: "Returns privacy policy article"
```

File: packages/frontend-api/src/Resources/config/graphql-types/QueryDecorator.types.yaml [hunk 8]
```diff
@@ -83,3 +83,15 @@ QueryDecorator:
...
+            cookiesArticle:
+                type: 'Article'
+                resolve: "@=resolver('cookiesArticle')"
+                description: "Returns Cookies article"
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Resources/config/graphql-types/QueryDecorator.types.yaml:97]:
```suggestion
                description: "Returns information about cookies article"
```

File: project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php [hunk 9]
```diff
@@ -67,6 +102,39 @@ private function getArticleDataProvider(): array
...
+     */
+    public function getSpecialArticleDataProvider(): array
+    {
+        return [
+            [
+                'termsAndConditionsArticle',
+                [
+                    'name' => 'Terms and conditions',
+                    'placement' => Article::PLACEMENT_FOOTER,
+                    'text' => 'Morbi posuere mauris dolor, quis accumsan dolor ullamcorper eget. Phasellus at elementum magna, et pretium neque. Praesent tristique lorem mi, eget varius quam aliquam eget. Vivamus ultrices interdum nisi, sed placerat lectus fermentum non. Phasellus ac quam vitae nisi aliquam vestibulum. Sed rhoncus tortor a arcu sagittis placerat. Nulla lectus nunc, ultrices ac faucibus sed, accumsan nec diam. Nam auctor neque quis tincidunt tempus. Nunc eget risus tristique, lobortis metus vitae, pellentesque leo. Vivamus placerat turpis ac dolor vehicula tincidunt. Sed venenatis, ante id ultrices convallis, lacus elit porttitor dolor, non porta risus ipsum ac justo. Integer id pretium quam, id placerat nulla.',
```
Comment (TomasLudvik) [inline project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php:116]: Why are you providing only some data when you are querying all data in getSpecialArticleQuery method?

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php (hunk 9 above)
Comment (s3tezsky) [reply project-base/tests/FrontendApiBundle/Functional/Article/GetArticleTest.php:123]: fixed

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php (hunk 5 above)
Comment (s3tezsky) [reply packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php]: changed

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php (hunk 3 above)
Comment (s3tezsky) [reply packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php]: changedd

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php (hunk 4 above)
Comment (s3tezsky) [reply packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php]: changedd

File: packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php [hunk 10]
```diff
@@ -25,16 +27,32 @@ class ArticleResolver implements ResolverInterface, AliasedInterface
...
+    protected $cookiesFacade;
+
     /**
      * @param \Shopsys\FrameworkBundle\Model\Article\ArticleFacade $articleFacade
      * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Model\LegalConditions\LegalConditionsFacade $legalConditionsFacade
+     * @param \Shopsys\FrameworkBundle\Model\Cookies\CookiesFacade $cookiesFacade
      */
     public function __construct(
         ArticleFacade $articleFacade,
```
Comment (TomasLudvik) [inline packages/frontend-api/src/Model/Resolver/Article/ArticleResolver.php:47]: This is not in use anymore

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

**Review by TomasLudvik — APPROVED** [review]

**Review by sslt — APPROVED** [review]

**Pull request merged**